sdk/identity/azure-identity/azure/identity/_internal/interactive.py

```diff
@@ -140,23 +140,68 @@ def get_token(
             configured not to begin this automatically. Call :func:`authenticate` to begin interactive authentication.
         """
+        options: TokenRequestOptions = {}
+        if claims:
+            options["claims"] = claims
+        if tenant_id:
+            options["tenant_id"] = tenant_id
+        options["enable_cae"] = enable_cae
+
+        token_info = self._get_token_base(*scopes, options=options, base_method_name="get_token", **kwargs)
+        return AccessToken(token_info.token, token_info.expires_on)
+
+    def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] = None) -> AccessTokenInfo:
+        """Request an access token for `scopes`.
+
+        This is an alternative to `get_token` to enable certain scenarios that require additional properties
+        on the token. This method is called automatically by Azure SDK clients.
+
+        :param str scopes: desired scopes for the access token. This method requires at least one scope.
+            For more information about scopes, see https://learn.microsoft.com/entra/identity-platform/scopes-oidc.
+        :keyword options: A dictionary of options for the token request. Unknown options will be ignored. Optional.
+        :paramtype options: ~azure.core.credentials.TokenRequestOptions
+
+        :rtype: AccessTokenInfo
+        :return: An AccessTokenInfo instance containing information about the token.
+
+        :raises CredentialUnavailableError: the credential is unable to attempt authentication because it lacks
+            required data, state, or platform support
+        :raises ~azure.core.exceptions.ClientAuthenticationError: authentication failed. The error's ``message``
+            attribute gives a reason.
+        :raises AuthenticationRequiredError: user interaction is necessary to acquire a token, and the credential is
+            configured not to begin this automatically. Call :func:`authenticate` to begin interactive authentication.
+        """
+        return self._get_token_base(*scopes, options=options, base_method_name="get_token_info")
+
+    def _get_token_base(
+        self,
+        *scopes: str,
+        options: Optional[TokenRequestOptions] = None,
+        base_method_name: str = "get_token_info",
+        **kwargs: Any,
+    ) -> AccessTokenInfo:
         if not scopes:
-            message = "'get_token' requires at least one scope"
-            _LOGGER.warning("%s.get_token failed: %s", self.__class__.__name__, message)
+            message = f"'{base_method_name}' requires at least one scope"
+            _LOGGER.warning("%s.%s failed: %s", self.__class__.__name__, base_method_name, message)
             raise ValueError(message)
 
         allow_prompt = kwargs.pop("_allow_prompt", not self._disable_automatic_authentication)
+        options = options or {}
+        claims = options.get("claims")
+        tenant_id = options.get("tenant_id")
+        enable_cae = options.get("enable_cae", False)
```

Comment:
This will lose all other options (e.g. pop), right?

Reply:
Updated to propagate additional options to support the pop case in the broker credential. So now `_request_token` in the broker credential will now get the `pop` kwarg if `pop` was passed through the options.

Also updated chained `credential` type to be `TokenProvider`, and added chained support for credentials that only have `get_token_info` implemented.